src/vm/method.cpp

```diff
@@ -4959,84 +4959,80 @@ LPVOID NDirectMethodDesc::FindEntryPoint(HINSTANCE hMod) const
     
-    FARPROC pFunc = NULL, pFuncW = NULL;
+    FARPROC pFunc = NULL;
 
 #ifndef FEATURE_PAL
     // Handle ordinals.
-    if (GetEntrypointName()[0] == '#')
+    if (funcName[0] == '#')
     {
-        long ordinal = atol(GetEntrypointName()+1);
+        long ordinal = atol(funcName + 1);
         return reinterpret_cast<LPVOID>(GetProcAddress(hMod, (LPCSTR)(size_t)((UINT16)ordinal)));
     }
 #endif
 
+    BOOL onlyCheckUserProvidedName = IsNativeNoMangled();
+
+#if defined(_TARGET_X86_)
+    onlyCheckUserProvidedName = onlyCheckUserProvidedName && !IsStdCall();
+#endif
+
     // Just look for the unmangled name.  If it is unicode fcn, we are going
     // to need to check for the 'W' API because it takes precedence over the
     // unmangled one (on NT some APIs have unmangled ANSI exports).
-    pFunc = GetProcAddress(hMod, funcName = GetEntrypointName());
-    if ((pFunc != NULL && IsNativeAnsi()) || IsNativeNoMangled())
+    pFunc = GetProcAddress(hMod, funcName);
+    if ((pFunc != NULL && IsNativeAnsi()) || onlyCheckUserProvidedName)
     {
-        return (LPVOID)pFunc;
+        return reinterpret_cast<LPVOID>(pFunc);
     }
 
     // Allocate space for a copy of the entry point name.
-    int dstbufsize = (int)(sizeof(char) * (strlen(GetEntrypointName()) + 1 + 20)); // +1 for the null terminator
+    int dstbufsize = (int)(sizeof(char) * (strlen(funcName) + 1 + 20)); // +1 for the null terminator
                                                                          // +20 for various decorations
 
     // Allocate a single character before the start of this string to enable quickly
     // prepending a '_' character if we look for a stdcall entrypoint later on.
-    LPSTR szAnsiEntrypointName = ((LPSTR)_alloca(dstbufsize + 1)) + 1;
+    LPSTR szProbedEntrypointName = ((LPSTR)_alloca(dstbufsize + 1)) + 1;
 
     // Copy the name so we can mangle it.
-    strcpy_s(szAnsiEntrypointName,dstbufsize,GetEntrypointName());
-    DWORD nbytes = (DWORD)(strlen(GetEntrypointName()) + 1);
-    szAnsiEntrypointName[nbytes] = '\0'; // Add an extra '\0'.
-
+    strcpy_s(szProbedEntrypointName, dstbufsize, funcName);
+    DWORD probedFuncNameLength = (DWORD)(strlen(funcName) + 1);
+    szProbedEntrypointName[probedFuncNameLength] = '\0'; // Add an extra '\0'.
 
-    // If the program wants the ANSI api or if Unicode APIs are unavailable.
-    if (IsNativeAnsi())
+    if(!IsNativeNoMangled())
     {
-        szAnsiEntrypointName[nbytes-1] = 'A';
-        pFunc = GetProcAddress(hMod, funcName = szAnsiEntrypointName);
-    }
-    else
-    {
-        szAnsiEntrypointName[nbytes-1] = 'W';
-        pFuncW = GetProcAddress(hMod, szAnsiEntrypointName);
+        szProbedEntrypointName[probedFuncNameLength - 1] = IsNativeAnsi() ? 'A' : 'W';
 
-        // This overrides the unmangled API. See the comment above.
-        if (pFuncW != NULL)
+        FARPROC pProbedFunc = GetProcAddress(hMod, szProbedEntrypointName);
+
+        if (pProbedFunc != NULL)
```

Comment:
This check isn't needed. Just assign directly to `pFunc` instead of creating a local.

Reply:
This check actually is needed. In the case that the user-provided name resolves to a function and is marked Unicode, but the user-provided name + 'W' doesn't resolve.